docs/design.md

```diff
@@ -0,0 +1,118 @@
+# Overview
+
+Concord-bft is an implementation of the [SBFT
+Protocol](https://arxiv.org/abs/1804.01626). SBFT implements State Machine
+Replication (SMR) in the byzantine model. Concord-bft is a generic protocol,
+with pluggable interfaces for communication, storage, state transfer, and
+abstract state machine execution. This document covers the implementation
+of concord-bft. When the implementation differs from the paper, it will be
+mentioned explicitly.
+
+It is important to note that this document covers the code as it exists at the
+time of writing. When the code changes substantially this document should be
+updated.
+
+# Public Interfaces
+
+Concord-bft is a library meant to be used by application code. The external user
```

Comment:
Concord-bft is a library used by the application code. The core engine API is described by a set of structs and abstract interfaces.